### PR TITLE
WIP: `jekyll doctor` should check for drafts outside custom collections directory

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adjusts the doctor command to check for drafts outside the custom collections directory

Follows https://github.com/jekyll/jekyll/pull/6985#issuecomment-386623372.

### CODE REVIEW

File: lib/jekyll/commands/doctor.rb [hunk 1]
```diff
@@ -39,18 +39,28 @@ def healthy?(site)
             !conflicting_urls(site),
             !urls_only_differ_by_case(site),
             proper_site_url?(site),
+            properly_gathered_drafts?(site),
             properly_gathered_posts?(site),
           ].all?
         end
 
+        def properly_gathered_drafts?(site)
+          inside_custom_collections_dir(site, "_drafts")
```
Comment (localheinz) [inline lib/jekyll/commands/doctor.rb]: @ashmaroli 

RubyMine complains:

screen shot 2018-05-04 at 16 54 07

**Comment by ashmaroli** [comment]
@localheinz
  - Git commit messages are best kept short (less than 50 characters). Please avoid unnecessary identifiers such as `Fix:`, `Enhancement:`, etc in the commit messages and PR titles. (*I've adjusted the message to demonstrate.*)
  -  `private` methods should be defined *after all the `public` methods* in a class. (*I've moved the `#inside_custom_collections_dir?` method for you to demonstrate.*)

File: lib/jekyll/commands/doctor.rb [hunk 2]
```diff
@@ -39,20 +39,17 @@ def healthy?(site)
             !conflicting_urls(site),
             !urls_only_differ_by_case(site),
             proper_site_url?(site),
+            properly_gathered_drafts?(site),
             properly_gathered_posts?(site),
           ].all?
         end
 
+        def properly_gathered_drafts?(site)
+          inside_custom_collections_dir?(site, "drafts")
```
Comment (ashmaroli) [inline lib/jekyll/commands/doctor.rb]: the directory name should start with an underscore (`_`)
```ruby
inside_custom_collections_dir?(site, "_drafts")
```
or handled appropriately in the private helper method..

Comment (localheinz) [reply lib/jekyll/commands/doctor.rb]: Ha, right. I’ll modify as soon as I’m back.

**Comment by ashmaroli** [comment]
@localheinz Will you be able to write a Cucumber test for this?

**Comment by localheinz** [comment]
@ashmaroli 

I’ll try! 👍

**Pull request closed without merging**